crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -200,107 +197,85 @@ impl<'a> TypeScript<'a> {
                         new_stmts.push(transformed);
                     }
                 }
-                Statement::ClassDeclaration(decl) => {
+                Statement::ClassDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::ClassDeclaration(decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::FunctionDeclaration(decl) => {
+                Statement::FunctionDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::FunctionDeclaration(decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::TSEnumDeclaration(enum_decl) => {
+                Statement::TSEnumDeclaration(ref enum_decl) => {
                     is_empty = false;
                     names.insert(enum_decl.id.name.clone());
-                    new_stmts.push(Statement::TSEnumDeclaration(enum_decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::VariableDeclaration(decl) => {
+                Statement::VariableDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::VariableDeclaration(decl));
+                    new_stmts.push(stmt);
+                }
+                Statement::ExportNamedDeclaration(ref export_decl)
+                    if export_decl.declaration.is_none() =>
+                {
+                    // TODO: This isn't correct. Need to transform `export {x}` somehow.
+                    new_stmts.push(stmt);
```

Comment:
@Dunqing This PR doesn't change the logic, but this doesn't look right to me. We can end up with `export {x}` in the output. [Playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICsgICAgICAgIC3mMl9fWP2DOEtS29bAAVF3HDuYFci2KI8dKJg9t1AO3n4jQYRoCBjAGTVEvqeRiV%2FeJWQgA%3D%3D)

I'm not sure what the correct behaviour is, though.

Reply:
You are right. We don't worry about it. This is not valid code in typescript https://www.typescriptlang.org/play/?verbatimModuleSyntax=true#code/BQMwrgdgxgLglgewgKmASgAQG8BQH8bjTxLAAeGAvBgJ5wCmANgCYYDMmWAvjl2jkA

We can also add a check for `export {x}` inside the namespace in https://github.com/oxc-project/oxc/blob/05-29-refactor_typescript_namespace_reuse_tsmoduleblock_s_scope_id/crates/oxc_semantic/src/checker/typescript.rs#L169

Reply:
So can we just remove this branch entirely? Do nothing rather than `new_stmts.push(stmt)`?

Reply:
Yes, we can.

Reply:
Thanks for coming back. I'll make that change before we merge this.

Reply:
That branch removed in #3475.